crates/ibc/src/core/ics02_client/consensus_state.rs

```diff
@@ -55,6 +55,15 @@ pub fn downcast_consensus_state<CS: ConsensusState>(h: &dyn ConsensusState) -> O
     h.as_any().downcast_ref::<CS>()
 }
 
+impl TryFrom<Box<dyn ConsensusState>> for Vec<u8> {
+    type Error = ClientError;
+
+    fn try_from(value: Box<dyn ConsensusState>) -> Result<Self, Self::Error> {
+        let value = value.encode_vec().map_err(ClientError::Encode)?;
+        Ok(value)
+    }
+}
+
```

Comment:
same here, this implementation is just one of many ways to serialize a `Box<dyn ConsensusState>`, and so shouldn't live here

Reply:
fcd8b791d29a82e838e7724c58ae4e804502bc29